### PR TITLE
fix/persisting-unique-id-on-db-connector

### PR DESCRIPTION
**Pull request by kirtimanmishrazipstack**
## What

- Adding unique id in db connector
- Handle error scenarios in-case column doesn't exists for user connector databases

## Why

- To uniquely identify connector data

## How

- Adding `execution_id` and `file_path` to output db connector
- Add error scenarios if  column doesn't exists

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- NO

## Database Migrations

-  NO

## Env Config

- NO

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

- Tests workflow with the changes in db connector
- Checked error scenarios for all connectors

## Screenshots
Screenshot from 2024-07-23 08-23-31
Screenshot from 2024-07-23 08-23-27
Screenshot from 2024-07-23 15-13-07
Screenshot from 2024-07-25 13-54-15

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/workflow_manager/endpoint/constants.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class TableColumns:
     CREATED_BY = "created_by"
     CREATED_AT = "created_at"
-    PERMANENT_COLUMNS = ["created_by", "created_at"]
+    PERMANENT_COLUMNS = ["created_by", "created_at", "file_path", "execution_id"]
```
Comment (chandrasekharan-zipstack) [inline backend/workflow_manager/endpoint/constants.py:4]: @kirtimanmishrazipstack we agreed that we'll allow these column names to be modifiable by the user similar to the `data` column. Can you make that change?

Comment (kirtimanmishrazipstack) [reply backend/workflow_manager/endpoint/constants.py]: @chandrasekharan-zipstack done

File: backend/workflow_manager/endpoint/static/dest/db.json [hunk 2]
```diff
@@ -36,6 +36,18 @@
...
+            "title": "File Path",
+            "default": "file_path",
+            "description": "File path to store the file address"
+        },
+        "executionId": {
+            "type": "string",
+            "title": "Execution Id",
+            "default": "execution_id",
+            "description": "Execution id of the workflow"
+        },
```
Comment (chandrasekharan-zipstack) [inline backend/workflow_manager/endpoint/static/dest/db.json:50]:
```suggestion
        "filePath": {
            "type": "string",
            "title": "File Path Column Name",
            "default": "file_path",
            "description": "Name of the column to store the absolute path to the file"
        },
        "executionId": {
            "type": "string",
            "title": "Execution ID Column Name",
            "default": "execution_id",
            "description": "Name of the column that stores the execution ID which identifies each individual run of a workflow"
        },
```

**Review by chandrasekharan-zipstack — APPROVED** [review]

File: backend/workflow_manager/endpoint/static/dest/db.json (hunk 2 above)
Comment (kirtimanmishrazipstack) [reply backend/workflow_manager/endpoint/static/dest/db.json]: @chandrasekharan-zipstack changes done

**Comment by athul-rs** [comment]
@kirtimanmishrazipstack @chandrasekharan-zipstack One doubt here, will this cause any issue for previous data stored in db, assuming here we are now adding `input_file_path` which won't be present for prior data ?

```
            self.insert_into_db(
                file_history=file_history, input_file_path=input_file_path
            )
```

**Comment by johnyrahul** [comment]
@kirtimanmishrazipstack  FE screenshot seems to be outdated?

**Comment by kirtimanmishrazipstack** [comment]
> insert_into_db



> @kirtimanmishrazipstack FE screenshot seems to be outdated?

@johnyrahul new FE changes are yet to come. This backend change will comprehend the new FE changes too. I have just changes `db.json`

**Comment by kirtimanmishrazipstack** [comment]
> @kirtimanmishrazipstack @chandrasekharan-zipstack One doubt here, will this cause any issue for previous data stored in db, assuming here we are now adding `input_file_path` which won't be present for prior data ?
> 
> ```
>             self.insert_into_db(
>                 file_history=file_history, input_file_path=input_file_path
>             )
> ```

@athul-rs I have just used `input_file_path` to store in db. Not tinkered `file_history`. It should not break anything.

**Comment by kirtimanmishrazipstack** [comment]
@johnyrahul @athul-rs @chandrasekharan-zipstack 
As discussed we decided to not go for any migration on the user table. The user can create a separate table and use the db-connector

**Comment by chandrasekharan-zipstack** [comment]
Thanks for pointing it out @athul-rs @johnyrahul.
@kirtimanmishrazipstack what will the behaviour be for existing users who use this connector though? Ideally it should either just work for existing tables or if an error is thrown, we should clearly mention that its because these columns are missing and they need to add it. Please ensure this is checked / handled as well

**Review by chandrasekharan-zipstack — APPROVED** [review]

**Review by johnyrahul — APPROVED** [review]

**Pull request merged**